solver/src/solver/http_solver/model.rs

```diff
@@ -1,20 +1,21 @@
+use ethcontract::H160;
 use model::u256_decimal;
 use primitive_types::U256;
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
 #[derive(Debug, Default, Serialize)]
 pub struct BatchAuctionModel {
-    pub tokens: HashMap<String, TokenInfoModel>,
+    pub tokens: HashMap<H160, TokenInfoModel>,
     pub orders: HashMap<String, OrderModel>,
     pub uniswaps: HashMap<String, PoolModel>,
     pub metadata: Option<MetadataModel>,
 }
 
 #[derive(Debug, Serialize)]
 pub struct OrderModel {
-    pub sell_token: String,
-    pub buy_token: String,
+    pub sell_token: H160,
+    pub buy_token: H160,
```

Comment:
This is likely incorrect because it changes how this is formatted in the json. token_to_string says that the string has to start with with a letter so we need a custom serialization function that does this here too.

Reply:
But there isn't actually a good reason why the string has to start with the letter `t` is there?

Reply:
Anyway, this is just here in case the solver team decided they would like to start accepting tokens by address instead of some arbitrary index denoted as `t_i`. we don't have to merge it, but its ready to go and addresses the totally valid point brought up in #761.

Reply:
We can have an opaque `TokenId(String)` type if we want to stay 100% backwards compatible. The comment in the other PR is that `String` is just textual data and doesn't give any indication as to what it represents:
- `42.to_string()`
- `"asdf"`
- `token_to_string(token)`

Are all strings, but have vastly different meaning. My point in the other PR is that we could use more expressive types to clear what the data represents.

From the other PR:
> I have learned that the String we are using for tokens is actually just their index in an array

This is a strong argument for changing the types, as from looking at the type definition you weren't able to see what the expected data was. In fact, it would be super easily to accidentally insert `token_address.to_string()` instead of `token_index.to_string()` in some code and generate a bad instance without having the compiler tell you that it was incorrect.

Reply:
Cool. So I see now that we have a choice to make. 

`TokenId(String)` where we continue to send tokens identified as `tN` or continue with the refactoring as currently found in this PR.

My argument for the latter is that we are already making breaking changes on the solver side so this would be the most appropriate time to introduce the more natural token identifier as H160.